src/components/addstudent.tsx

```diff
@@ -1,25 +1,34 @@
 import * as React from "react";
 
-import { Container, Table, Button, Icon, Modal, Form, Header, Image, Input, Grid, Comment, TextArea} from "semantic-ui-react";
+import { Container, Table, Button, Icon, Modal, Form, Header, Image, Input, Grid, Comment, TextArea, Popup } from "semantic-ui-react";
 
 export interface AddStudentProps {
     courseLink?: string
 }
 
 export interface AddStudentState {
-    isModalOpen: boolean
+    isModalOpen: boolean,
+    showLinkCopiedMessage: boolean
 }
 
 export class AddStudent extends React.Component<AddStudentProps, AddStudentState> {
     constructor(props: AddStudentProps) {
         super(props);
 
         this.state = { 
-            isModalOpen: false
+            isModalOpen: false,
+            showLinkCopiedMessage: false
         };
     }
 
     copyCourseLink() {
+        const timeoutLength = 5 * 1000;
+        this.setState({ showLinkCopiedMessage: true });
+
+        setTimeout(() => {
```

Comment:
use requestAnimationFrame instead
see (here)[https://stackoverflow.com/questions/38709923/why-is-requestanimationframe-better-than-setinterval-or-settimeout] why it's better

Reply:
Yeah, but this code does not have an intent to animate things (what's more important - *manually* animate). Thus I believe `requestAnimationFrame` is a bad candidate here...